src/bartiq/compilation/types.py

```diff
@@ -12,12 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Callable, Union
+import math
+from typing import Callable, Dict, Union
 
 # Note: when we drop support for 3.9, we will be able to start
 # using the Number union with isinstance checks, thus eliminating
 # the need of separate NUMBER_TYPES
 NUMBER_TYPES = (int, float)
 Number = Union[int, float]
 
+# Define mathematical constants
+Math_constants: Dict[str, float] = {
+    "pi": math.pi,
```

Comment:
[issue] Actually `math.pi` returns a float value – what we wanted to have is a symbolic representation of `pi`, e.g.: `sympy.pi`.